ibis/backends/pandas/client.py

```diff
@@ -133,7 +168,19 @@ def infer_pandas_timestamp(value):
 
 @dt.infer.register(np.ndarray)
 def infer_array(value):
-    # TODO(kszucs): infer series
+    np_dtype_name = value.dtype.name
+    if np_dtype_name == 'object':
+        # The dtype of the np.array is ambiguous, so we need to infer the type
+        # of the elements manually.
+        if value.size == 0:
+            return dt.Array(dt.null)
+        return dt.Array(dt.highest_precedence(map(dt.infer, value)))
```

Comment:
I am not sure we want to infer all elements here. Can we use https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.api.types.infer_dtype.html instead?

Reply:
```
In [18]: pd.api.types.infer_dtype([1, 2.0])                                                                                                                                                                                                                   
Out[18]: 'mixed-integer-float'

In [19]: pd.api.types.infer_dtype(np.array([1, 2.0]))                                                                                                                                                                                                         
Out[19]: 'floating'

In [20]: pd.api.types.infer_dtype(pd.Series([1, 2.0]))                                                                                                                                                                                                        
Out[20]: 'floating'
```

Reply:
This seems to work as well
```
In [22]: pd.api.types.infer_dtype([])                                                                                                                                                                                                                         
Out[22]: 'empty'
```

Reply:
I've made this change—using a mapping of outputs from `pd.api.types.infer_dtype` -> Ibis dtypes

The inferred types will be more "coarse" (before, an array with small integers could be inferred to be `dt.Array(int8)`, but now would be `dt.Array(int64)`), but now we don't have to infer the type of every element in the array

Reply:
Sounds good. This is fine.